openedx/core/djangoapps/content/course_overviews/models.py

```diff
@@ -358,3 +363,21 @@ def get_all_course_keys(cls):
             CourseKey.from_string(course_overview['id'])
             for course_overview in CourseOverview.objects.values('id')
         ]
+
+    def is_discussion_tab_enabled(self):
+        """
+        Returns True if course has discussion tab and is enabled
+        """
+        course_overview_tabs = self.course_overview_tab.all()
+        for tab in course_overview_tabs:
+            if tab.tab_id == "discussion" and is_discussion_enabled(unicode(self.id)):
```

Comment:
`is_discussion_enabled` expects a `CourseKey` not a string.  So remove the `unicode` conversion.